enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+
+[^8]: To allow continued operation of MachineSets, ControlPlaneMachineSets and MachineHealthChecks, if a Machine API Machine that is owned is deleted,
+then the deletion event is synchronised to Cluster API.
+This will allow for a MachineSet/ControlPlaneMachineSet to own mixed instances (i.e. those managed by both Machine API and Cluster API), and continue to support higher level remediation automation.
+To remove the Machine API resources, the user should delete the parent, i.e. the MachineSet or ControlPlaneMachineSet on the Machine API side.
+
+#### Scenarios with mixed Machines
+
+The following scenarios explain the logical flow of what must happen if any of the controllers are monitoring a selection of Machines where the authoritative API is mixed between Cluster API and Machine API.
+
+When the selection of Machines all match the same API, the normal flows will be observed.
+
+##### MachineSet scale up
+
+If a MachineSet scales up, it will create a new Machine as it does today.
+Because the managed resource in this scenario is the MachineSet the synchronisation controller will ensure that the Machine resources are reflected in both Machine API and Cluster API.
+
+If the MachineSet is a Cluster API MachineSet and there is no Machine API equivalent, this synchronisation will not happen and the machines will be created as Cluster API only.
+
+New Machines created in the scale up with default to the same authoritative API as the MachineSet that creates them.
+
+##### MachineSet scale down
+
+If a MachineSet scales down, this will trigger the deletion of a Machine resource
+To ensure consistency between the two MachineSets, the synchronisation controller will synchronise the delete event between the Cluster API and Machine API Machines.
+
+##### MachineHealthCheck remediates Machine
+
+As MachineHealthChecks only remediate Machines that have owners, when the deletion event occurs, the deletion event will be synchronised across between the two APIs.
+
+##### Resource created as Cluster API Resource, wish to migrate to Machine API Resource
+
+To enable a reverse migration, should the customer wish, we will need to enable the user to create a Machine API equivalent of a Cluster API resource if it does not exist.
+
+The user will be able to do this by creating the Machine API equivalent resource and ensuring that the authoritative annotation is present from creation and is set to `cluster.x-k8s.io`.
+This will trigger the sync controller to copy the spec/status from the Cluster API resource onto the Machine API resource.
+
+At this point, the customer can switch back the annotation to `machine.openshift.io` to allow the Machine API controllers to take over the management of the resources.
+
+Creating the Machine in Machine API with the authoritative API set to `machine.openshift.io` will be considered an error and the API call will be rejected.
```

Comment:
Could we assume here that the user is trying to migrate? Or are you purposefully requiring them to make a stub MAPI resource with `cluster.x-k8s.io`, wait for a sync, then edit the object again as a way of confirming intentions?

I would agree that creating a Machine in MAPI with the authoritative API set to `machine.openshift.io` on creation would be a complicated case for us to manage, but I wonder how users would receive the currently presented workflow.

Reply:
The workflow here was for a couple of reasons, one to avoid having extra complication in the sync logic to handle this, the second, while yes, it's not ideal from an end user perspective, it does make it a very deliberate action to do this.
I think given the feedback we can give the user via a webhook/API validation, this should be relatively straight forward for them to correct their action should they see they got it "wrong"